code/datums/components/crafting/guncrafting.dm

```diff
@@ -1,22 +1,165 @@
+
+// PRIMARY //
+// Primary attachments are meant to affect the mechanical aspect of a weapon, for now, since im working on bows, im using this to separate bow string from other attachments
+// knowing this can be usefull for the future when its expanded into other weapons
+
+/obj/item/weaponcrafting/attachment/primary/cablestring
+	name = "cable coil string"
+	desc = "A cable coil string to be used in weapon crafting."
+	icon_state = "cablestring"
+	damage_multiplier = 0.5
+	speed_multiplier = 2
+	gun_suitable = FALSE
+	//Half damage and Half speed
+	added_description = "<span class='info'>The bowstring is improvised out of cable. It looks rather weak.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/bamboostring
+	name = "bamboo fiber string"
+	desc = "A string made out of bamboo fiber to be used in weapon crafting."
+	icon_state = "bamboostring"
+	damage_multiplier = 0.8
+	speed_multiplier = 0.8
+	gun_suitable = FALSE
+	//A bit weak and a bit fast
+	added_description = "<span class='info'>The bowstring is made of bamboo fiber. Close to standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/silkstring
 	name = "silkstring"
-	desc = "A long piece of Silk that looks like a cable coil."
-	icon = 'icons/obj/improvised.dmi'
+	desc = "A string made out of silk to be used in weapon crafting."
 	icon_state = "silkstring"
+	damage_multiplier = 1
+	speed_multiplier = 1
+	gun_suitable = FALSE
+	//Standard stuff, no debuffs
+	added_description = "<span class='info'>The bowstring is made of silkstring. Standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/leatherstring
+	name = "leather string"
+	desc = "A string made out of leather to be used in weapon crafting."
+	icon_state = "leatherstring"
+	damage_multiplier = 1
+	speed_multiplier = 1.2
+	gun_suitable = FALSE
+	//Slight debuff to speed
+	added_description = "<span class='info'>The bowstring is made of leather. Standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/sinewstring
+	name = "sinew string"
+	desc = "A bow string made out of sinew to be used in weapon crafting."
+	icon_state = "sinewstring"
+	damage_multiplier = 1.2
+	speed_multiplier = 1.2
+	gun_suitable = FALSE
+	//Buff to damage, debuff to speed
+	added_description = "<span class='info'>The bowstring is made of sinew. It looks pretty strong.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/energy_crystal //Not aviable ingame yet
+	name = "energy crystal"
+	desc = "An energy crystal made out of uranium used in the construction of energy weaponry. A warning lable reads 'Warning: Do Not Ingest'."
+	custom_materials = list(/datum/material/uranium = MINERAL_MATERIAL_AMOUNT * 1)
+	material_flags = MATERIAL_EFFECTS
+	icon_state = "crystal"
+	damage_multiplier = 1.5
+	speed_multiplier = 0.6
+	fire_sound = 'sound/weapons/edagger.ogg'
+	//Stronger, faster
+	added_description = "<span class='info'>The bowstring is made of pure energy. As robust as it gets.</span>"
```

Comment:
If this isn't available yet as part of this PR, it shouldn't be included here.

Reply:
Very well, I'll save this for later, as well as all of its moving parts. However I will only be taking them off once im sure no changes to the base mechanics of the bows will be necessary, as to maintain that part of the code (energy bow) functional.